Entities/Structures/Components/Passive/Magazine/Magazine.as

```diff
@@ -65,47 +65,45 @@ void GetButtonsFor(CBlob@ this, CBlob@ caller)
 
 	if (this.getDistanceTo(caller) > 16.0f || !this.getShape().isStatic()) return;
 
-	CBlob@ carried = caller.getAttachments().getAttachmentPointByName("PICKUP").getOccupied();
-	if (carried !is null)
-	{
-		if (carried.getRadius() > 3.5f || !carried.canBePutInInventory(this)) return;
-
-		CBlob@ item = this.getInventory().getItem(0);
-		if (item !is null && (item.getName() != carried.getName() || item.getQuantity() == item.maxQuantity)) return;
-
-		CBitStream params;
-		params.write_u16(carried.getNetworkID());
+	CBlob@ carried = caller.getCarriedBlob();
+	bool LOAD = (carried !is null);
 
-		CButton@ button = caller.CreateGenericButton(
-		"$"+carried.getName()+"$",              // icon token
-		Vec2f_zero,                             // button offset
-		this,                                   // button attachment
-		this.getCommandID("load"),              // command id
-		getTranslatedString("Load"),            // description
-		params);                                // cbitstream
+	CBlob@ item = this.getInventory().getItem(0);
 
-		button.radius = 8.0f;
-		button.enableRadius = 22.0f;
+	if (LOAD)
+	{
+		if  (!carried.canBePutInInventory(this) 			// doesn't go in inventories to begin with
+			|| !this.getInventory().canPutItem(carried)  	// does go in inventories but doesn't fit
+			|| (item !is null && (item.getName() != carried.getName() || item.getQuantity() == item.maxQuantity)))
+		{
+			return;
+		}
 	}
-	else
+	else if (item is null)
 	{
-		CBlob@ item = this.getInventory().getItem(0);
-		if (item is null) return;
-
-		CBitStream params;
-		params.write_u16(caller.getNetworkID());
-
-		CButton@ button = caller.CreateGenericButton(
-		"$"+item.getName()+"$",                 // icon token
-		Vec2f_zero,                             // button offset
-		this,                                   // button attachment
-		this.getCommandID("unload"),            // command id
-		getTranslatedString("Unload"),          // description
-		params);                                // cbitstream
+		return;
+	}
 
-		button.radius = 8.0f;
-		button.enableRadius = 22.0f;
+	CBitStream params;
+	params.write_u16(LOAD ? carried.getNetworkID() : caller.getNetworkID());
+		
+	string iconName = LOAD ? "$" + carried.getName() + "$" : "$" + item.getName() + "$";
+	if (LOAD ? carried.hasTag("use inventory icon") : item.hasTag("use inventory icon"))
+	{
+		iconName = LOAD ? "$" + carried.getInventoryName() + "$" : "$" + item.getInventoryName() + "$";
 	}
+
+	CButton@ button = caller.CreateGenericButton(
+	iconName, 																									// icon token
+	Vec2f_zero,																									// button offset
+	this, 																										// button attachment
+	this.getCommandID(LOAD ? "load" : "unload"), 																// command id
+	LOAD ? getTranslatedString("Load {ITEM}").replace("{ITEM}", carried.getInventoryName())
+		 : getTranslatedString("Unload {ITEM}").replace("{ITEM}", item.getInventoryName()), 					// description
```

Comment:
Fix indentation here as well (make : line up with ?)

Reply:
It looks lined up for me: https://imgur.com/JOkn8H5.png
Should I be doing this? https://i.imgur.com/gEJUI6e.png

Reply:
Nvm that doesn't matter it only shows up like that on GitHub